sympy/physics/hep/tests/test_gamma_matrices.py

```diff
@@ -399,3 +400,25 @@ def test_gamma_matrix_trace():
     t = ps*ps*ps*ps*ps*ps*ps*ps
     r = gamma_trace(t)
     assert r.equals(4*p2*p2*p2*p2)
+
+def test_bug_13636():
+    """Test issue 13636 regarding handling traces of sums of products
+    of GammaMatrix mixed with other factors."""
+    pi, ki, pf = tensor_heads("pi, ki, pf", [LorentzIndex])
+    i0, i1, i2, i3, i4 = tensor_indices("i0:5", LorentzIndex)
+    x = Symbol("x")
+    pis = pi(i2) * G(-i2)
+    kis = ki(i3) * G(-i3)
+    pfs = pf(i4) * G(-i4)
+
+    a = pfs * G(i0) * kis * G(i1) * pis * G(-i1) * kis * G(-i0)
+    b = pfs * G(i0) * kis * G(i1) * pis * x * G(-i0) * pi(-i1)
+    ta = gamma_trace(a)
+    tb = gamma_trace(b)
+    t_a_plus_b = gamma_trace(a + b)
+    assert ta.equals(
+        -16 * ki(i0) * ki(-i0) * pf(i1) * pi(-i1)
+        + 32 * ki(i0) * ki(i1) * pf(-i0) * pi(-i1)
+    )
+    assert tb.equals(-8 * x * ki(i0) * pf(-i0) * pi(i1) * pi(-i1))
+    assert t_a_plus_b.equals(ta + tb)
```

Comment:
Does `==` not work here or does it need to use `.equals`?

Reply:
Good question. I just checked, and indeed == does not work here. The .equals() function for tensor objects applies sympify and .cannon_bp to the tensor expression before comparing, while ._eq_ is just Basic._eq_, and is false for this comparison because the expression trees are not exactly identical. The other test functions such as test_gamma_matrix_trace() also use .equals().

Reply:
Usually it is better to construct the exact expression for comparison in tests so that they can be compared with `==`. Sometimes that is not be best option because the exact expression is confusing and does not display the intent of the test well.

I'm not sure what the preferred answer would be here. What exactly is the difference between the compared expressions? You can use `srepr` as one way to view the expression structure explicitly.

Reply:
ta == 4*((-4)*ki(L_0)*ki(-L_0)*pf(L_1)*pi(-L_1) + 8*ki(L_0)*ki(L_1)*pf(-L_0)*pi(-L_1)) which is the same as the expression being tested against except for distribution of the factor of 4 over the coefficients. I can use the exact expression for the test, I don't mind.

tb and ta_plus_tb are the same as the expressions they are being compared against, so we could use == there with no problem.

Reply:
This is fixed now.